src/rest/connector/libs/nd/implementation.py

```diff
@@ -0,0 +1,310 @@
+import json
+import logging
+import requests
+import time
+import urllib.parse
+from inspect import signature
+
+from requests.exceptions import RequestException
+
+from pyats.connections import BaseConnection
+from rest.connector.implementation import Implementation as Imp
+from rest.connector.utils import get_username_password
+
+# create a logger for this module
+log = logging.getLogger(__name__)
+
+
+class Implementation(Imp):
+    """Rest Implementation for ND
+    Implementation of Rest connection to devices based on pyATS BaseConnection
+    for ND
+
+    YAML Example
+    ------------
+
+        devices:
+            nd1:
+                platform: nd
```

Comment:
I believe should be `os`. did you test?
```suggestion
                os: nd
```

Reply:
No, it's platform - the abstraction layer works on 'os' or 'platform' - 'os' is fine in most cases, but for Nexus Dashboard nodes (basically, linux-based virtual machines) 'os' should be linux, so platform is used for the abstraction.
Tested, it works as intended

Reply:
it's fine. iosxe is also based on linux. but os is `iosxe`. first abstraction should be done by os usually. so I would suggest to have `os: nd`.

Reply:
As far as i know yes, `nd` is the official short name.
I ran some tests, as follows:
the device uses 2 types of connections - a `rest` connection to the `nd` software and an `ssh` connection straight to the linux machine.
With abstraction set to `os` and `os: nd` the `ssh` connection failed (the `rest` one worked properly).
With abstraction set to `platform`, `os: linux` and `platform: nd` both connections work.
I have limited knowledge on this platform / pyats entirely, but i will ask my more knowledgeable colleagues if it's possible to go with `os: nd`

Reply:
I understand your point. currently, you need to have 2 devices for nd with different name like below in case you want to use 2 interfaces (CLI and REST)

```
devices:
  nd: # for CLI
    os: linux
  nd_test # for REST
    os: nd
```
if you really want to have only one device for both interfaces in testbed yaml, you might need to contribute Unicon.plugin for `nd`.

for example, `ise` plugin inherited `linux` plugin. something like this for `nd`.
https://github.com/CiscoTestAutomation/unicon.plugins/blob/master/src/unicon/plugins/ise/__init__.py

Reply:
Thank you for the help, i changed ```os``` to ```nd``` and will work on adding a plugin in Unicon.plugin as well.